### PR TITLE
Add python release build

### PR DESCRIPTION
**Pull request by wallies**

### CODE REVIEW

**Comment by wallies** [comment]
So what this does is adds SHAs as the github actions which is one of the steps the open security foundation. I used https://app.stepsecurity.io/ for this as this is the recommended approach by openssf. 

The other thing this does is add a publish step on workflow dispatch, so manual run. We can change it to be based on tags if you wish

**Comment by fulmicoton** [comment]
Thank you! let me see if I can find someone to review this

**Comment by kapilt** [comment]
Re action locking, I’d rather see use of open source for security purposes.. https://github.com/sethvargo/ratchet is one such alternative. Use of such tool should be in its own pr, no commingled. It also avoids polluting every single job with extraneous steps. I’ll also note while it is a good practice, it is very rare in the wild to see action locking.

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+          args: --release -o dist --universal2
+
+      - name: Upload wheels
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8  # 3.1.0
+        with:
+          name: wheels
+          path: dist
+
+      - name: Upload release binaries
+        uses: alexellis/upload-assets@eaab1472d5c2f0df3115cac81136f03e7e61f771
```
Comment (kapilt) [inline .github/workflows/publish.yaml:144]: Using the pypa organization’s upload action is preferred

**Comment by kapilt** [comment]
It feels odd doing automation on a single old version of python on a single set of architecture, ideally this would be a matrix build with arch support for at least x86_64 and aarch64

**Comment by kapilt** [comment]
> I used https://app.stepsecurity.io/ for this as this is the recommended approach by openssf.

could you clarify where you see/found this recommendation?, the only reference I can find to it, is a pr announce from step security itself.

**Comment by wallies** [comment]
> > I used https://app.stepsecurity.io/ for this as this is the recommended approach by openssf.
> 
> could you clarify where you see/found this recommendation?, the only reference I can find to it, is a pr announce from step security itself.

https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies

**Comment by Sidhant29** [comment]
@wallies is this PR good to go ?

**Comment by wallies** [comment]
@Sidhant29 no needs to pull across changes from Kapiche fork

**Comment by kapilt** [comment]
Lgtm, the tooling on step security is both helpful and easily replaced.

**Review by cjrh — APPROVED** [review]

**Pull request merged**